src/main/java/wolox/training/models/User.java

```diff
@@ -58,23 +57,23 @@ public String getUsername() {
     }
 
     private void setUsername(String username) {
-        this.username = username;
+        this.username = checkNotNull(username);
```

Comment:
After checking if a field is null, check if the field is empty.

An error message can be added to each check :

```
Preconditions.checkNotNull(username, "The username field cannot be null.");
Preconditions.checkArgument(!username.isEmpty(), "The username field cannot be empty.");
this.username = username;
```

I recommend using a class to store error messages as constants.